Sources/Datadog/RUM/RUMEvent/RUMEventEncoder.swift

```diff
@@ -8,22 +8,41 @@ import Foundation
 
 /// `Encodable` representation of RUM event.
 /// Mutable properties are subject of sanitization or data scrubbing.
-internal struct RUMEvent<DM: RUMDataModel>: Encodable {
+/// /// TODO: RUMM-1463 Remove `errorAttributes` property once new error format is managed through `RUMDataModels`
+/// TODO: RUMM-1584 - Remove `RUMEvent` container.
+internal struct RUMEvent<DM>: Encodable where DM: RUMDataModel, DM: RUMSanitizableEvent {
     /// The actual RUM event model created by `RUMMonitor`
     var model: DM
 
-    /// Custom attributes set by the user
-    var attributes: [String: Encodable]
-    var userInfoAttributes: [String: Encodable]
+    /// Error attributes.
+    let errorAttributes: [String: Encodable]?
```

Comment:
A handy note to future us could be this:
```suggestion
    /// Error attributes. Only set when `DM == RUMErrorEvent` and error describes a crash.
    /// Can be entirely removed when RUMM-1463 is resolved and error values are part of the `RUMErrorEvent`.
    let errorAttributes: [String: Encodable]?
```